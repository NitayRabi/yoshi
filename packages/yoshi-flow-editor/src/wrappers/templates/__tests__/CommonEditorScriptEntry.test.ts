import commonEditorScriptEntry from '../CommonEditorScriptEntry';

describe('CommonEditorScriptEntry template', () => {
  it('generates correct template with entry editorScript file for OOI components', () => {
    const generateEditorScriptEntryContent = commonEditorScriptEntry({
      editorEntryFileName: 'project/src/editor.app.ts',
      controllersMeta: [],
      shouldUseAppBuilder: false,
    });

    expect(generateEditorScriptEntryContent).toMatchSnapshot();
  });

  it('generates correct template with entry editorScript file for app builder components', () => {
    const generateEditorScriptEntryContent = commonEditorScriptEntry({
      editorEntryFileName: 'project/src/editor.app.ts',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/a/editor.controller.ts',
          id: 'FIRST_ID',
          widgetType: 'STUDIO_WIDGET',
        },
      ],
      shouldUseAppBuilder: true,
    });

    expect(generateEditorScriptEntryContent).toMatchSnapshot();
  });

  it('generates correct template with entry editorScript file for multiple app builder components', () => {
    const generateEditorScriptEntryContent = commonEditorScriptEntry({
      editorEntryFileName: 'project/src/editor.app.ts',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/a/editor.controller.ts',
          id: 'FIRST_ID',
          widgetType: 'STUDIO_WIDGET',
        },
        {
          controllerFileName: 'project/src/components/c/editor.controller.ts',
          id: 'SECOND_ID',
          widgetType: 'STUDIO_WIDGET',
        },
        {
          controllerFileName: 'project/src/components/d/editor.controller.ts',
          id: 'THIRD_ID',
          widgetType: 'STUDIO_WIDGET',
        },
      ],
      shouldUseAppBuilder: true,
    });

    expect(generateEditorScriptEntryContent).toMatchSnapshot();
  });
});
