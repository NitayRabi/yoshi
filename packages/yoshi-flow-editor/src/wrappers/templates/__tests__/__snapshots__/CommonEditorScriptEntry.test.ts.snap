// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonEditorScriptEntry template generates correct template with entry editorScript file for OOI components 1`] = `
"
  var editorScriptEntry = require('project/src/editor.app.ts');

  

  module.exports = editorScriptEntry.default || editorScriptEntry;
"
`;

exports[`CommonEditorScriptEntry template generates correct template with entry editorScript file for app builder components 1`] = `
"
  var editorScriptEntry = require('project/src/editor.app.ts');

  
  var editorScriptBuilder = require('@wix/bob-widget-services').editorScriptBuilder;

  var builder = editorScriptBuilder();
  if (editorScriptEntry.editorReady) {
    builder = builder.withEditorReady(editorScriptEntry.editorReady);
  }
  if (editorScriptEntry.appManifest) {
    builder = builder.withAppManifest(editorScriptEntry.appManifest);
  }
  if (editorScriptEntry.eventHandler) {
    builder = builder.withEventHandler(editorScriptEntry.eventHandler);
  }
  builder = builder.withWidget(require(\\"project/src/components/a/editor.controller.ts\\"));
  editorScriptEntry = builder.build();
  

  module.exports = editorScriptEntry.default || editorScriptEntry;
"
`;

exports[`CommonEditorScriptEntry template generates correct template with entry editorScript file for multiple app builder components 1`] = `
"
  var editorScriptEntry = require('project/src/editor.app.ts');

  
  var editorScriptBuilder = require('@wix/bob-widget-services').editorScriptBuilder;

  var builder = editorScriptBuilder();
  if (editorScriptEntry.editorReady) {
    builder = builder.withEditorReady(editorScriptEntry.editorReady);
  }
  if (editorScriptEntry.appManifest) {
    builder = builder.withAppManifest(editorScriptEntry.appManifest);
  }
  if (editorScriptEntry.eventHandler) {
    builder = builder.withEventHandler(editorScriptEntry.eventHandler);
  }
  builder = builder.withWidget(require(\\"project/src/components/a/editor.controller.ts\\"));
  builder = builder.withWidget(require(\\"project/src/components/c/editor.controller.ts\\"));
  builder = builder.withWidget(require(\\"project/src/components/d/editor.controller.ts\\"));
  editorScriptEntry = builder.build();
  

  module.exports = editorScriptEntry.default || editorScriptEntry;
"
`;
